bridge/server/app.ts

```diff
@@ -309,9 +312,25 @@ function handleError(err: any, req: Request, res: Response, authType: AuthType):
   }
 
   printError(err);
-  log.info(`Response status: ${err.response?.status}`);
+  log.error(`Response status ${err.response?.status} for ${req.method} ${req.url}`);
 
   return err.response?.status || 500;
 }
 
+function filterHeaders(headers: IncomingHttpHeaders): IncomingHttpHeaders {
+  const filteredHeaders = { ...headers };
+  const denyList = ['cookies', 'user-agent'];
+  for (const item of denyList) {
+    // it is safe to delete because all properties are marked as optional ?
+    delete filteredHeaders[item];
+  }
+  // remove additional sec headers
+  Object.keys(headers).forEach((key) => {
+    if (key.startsWith('sec')) {
+      delete filteredHeaders[key];
+    }
+  });
+  return filteredHeaders;
+}
```

Comment:
You can try to be more functional here. Something like:
```ts
const record = {...headers};
const filtered = Object.entries(record)
  .filter(entry => !(entry[0].startsWith('sec') || ['cookies', 'user-agent'].includes(entry[0])))
  .reduce((prev, current) => ({...prev, [current[0]]: current[1]}), {});
return filtered;
```

Reply:
I think it makes very hard to follow the logic and purpose of the code.

Reply:
Then just add named intermediate steps:
```ts
const record = {...headers};
const removeSecurityRelatedHeaders = (entry) =>  !(entry[0].startsWith('sec') || ['cookies', 'user-agent'].includes(entry[0]));
const toRecord = (prev, current) => ({...prev, [current[0]]: current[1]});
const filtered = Object.entries(record)
  .filter(removeSecurityRelatedHeaders)
  .reduce(toRecord, {});
return filtered;
```

Reply:
I agree that this is more readable. However, the whole code base is built without a functional and I would propose to follow similar code pattern that are already there.